src/matcher_combinators/matchers.clj

```diff
@@ -1,31 +1,39 @@
 (ns matcher-combinators.matchers
   (:require [matcher-combinators.core :as core]))
 
-(defn equals-value
+(defn equals
   "Matcher that will match when the given value is exactly the same as the
   `expected`."
   [expected]
-  (core/->Value expected))
+  (cond
+    (sequential? expected) (core/->EqualsSeq expected)
+    (set? expected)        (core/->EqualsSet expected false)
+    (map? expected)        (core/->EqualsMap expected)
+    :else                  (core/->Value expected)))
```

Comment:
is there a reason to use multimethods instead of this type of cond'ing to do function dispatch based on type?

Reply:
I think `cond` seems fine for this purpose.